### PR TITLE
feat: add Codeberg presence

### PR DESCRIPTION
**Pull request by spifory**
## Description 
This pull requests adds a presence for Codeberg (https://codeberg.org), it adds presences for _almost_ every page on the site, it does not have presences for files, specific issues/PRs, and profile ?tab (or /-/) paths.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image
</details>

### CODE REVIEW

File: websites/C/Codeberg/presence.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+			forks: {
+				details: `Viewing ${repoTitle}'s forks`,
+			},
+			watchers: {
+				details: `Viewing ${repoTitle}'s watchers`,
+			},
+			stars: {
+				details: `Viewing ${repoTitle}'s stargazers`,
+			},
+		};
```
Comment (DarkVillager) [inline websites/C/Codeberg/presence.ts:192]: Why not just use queryselector like this:
`document.querySelector('.active.item')`

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/C/Codeberg/presence.ts (hunk 1 above)
Comment (spifory) [reply websites/C/Codeberg/presence.ts:192]: Well for most of these `*Pages` constants I'm not able to specify able to use that in the details (for instance, under `stars` I say `stargazers` and not `stars` so I can't really use querySelector like that)

Though I think the code you've written can be used as a replacement for the `if (pathname.includes(...))` conditions

File: websites/C/Codeberg/presence.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+				.textContent.trim();
+
+		if (repoAvatar) {
+			presenceData.largeImageKey = repoAvatar.src;
+			presenceData.smallImageKey = Assets.Logo;
+		} else presenceData.largeImageKey = Assets.Logo;
+
+		presenceData.buttons = [
+			{
+				label: "View repository",
```
Comment (theusaf) [inline websites/C/Codeberg/presence.ts:144]:
```suggestion
				label: "View Repository",
```

File: websites/C/Codeberg/presence.ts [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+			}
+		} else if (href.includes("/-/projects"))
+			presenceData.details = `Viewing ${profileName}'s projects`;
+		else if (href.includes("/-/packages"))
+			presenceData.details = `Viewing ${profileName}'s packages`;
+		else presenceData.details = `Viewing profile: ${name}`;
+
+		presenceData.buttons = [
+			{
+				label: "View profile",
```
Comment (theusaf) [inline websites/C/Codeberg/presence.ts:114]:
```suggestion
				label: "View Profile",
```

File: websites/C/Codeberg/presence.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+			presenceData.details = `Viewing ${orgName}'s projects`;
+		else if (href.includes("/-/packages"))
+			presenceData.details = `Viewing ${orgName}'s packages`;
+		else if (href.includes("/org"))
+			presenceData.details = `Viewing ${orgName}'s teams/members`;
+		else presenceData.details = `Viewing organization: ${orgName}`;
+
+		presenceData.buttons = [
+			{
+				label: "View organization",
```
Comment (theusaf) [inline websites/C/Codeberg/presence.ts:221]:
```suggestion
				label: "View Organization",
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**